lib/cequel/metal/data_set.rb

```diff
@@ -565,6 +567,22 @@ def consistency(consistency)
         end
       end
 
+      def page_size(page_size)
+        clone.tap do |data_set|
+          data_set.query_page_size = page_size
+        end
+      end
+
+      def paging_state(paging_state)
+        clone.tap do |data_set|
+          data_set.query_paging_state = paging_state
+        end
+      end
+
+      def next_paging_state
+        execute_cql(*cql).paging_state
+      end
```

Comment:
Does this mean that the query is run twice? Ie, once to figure the paging state and once to iterate over the results? We should extract `execute_cql(*cql)` in to a `#results` method that memoizes that operation so we only do it once per data set instance.